### ISSUE TITLE
[Bug] Video Player temp files fills up C drive Appdata folder in less than 2 hours

### ISSUE THREAD

**Issue by mossarelli** [owlboy/greatpug-public#227]
### Describe the Bug

During special events like St Patrick's day or Halloween, the video player that plays music in the world, has a tendency to fill up the C drive with a temp file in the Appdata folder.

Specifically: "halloween_mix-2024-final-v1-vrcss[1].dat" in:
C:\Users\Administrator\AppData\Local\Microsoft\Windows\INetCache\IE\somenumber

This slows down the computer and makes it prompt notifications like: "running low on disk space", and also makes the game crash more frequently.

This has been an issue since last year. Since the Halloween version a couple of days ago, multiple of my friends have had their game crash or been randomly disconnected from the instance in the world.

Take note of the free space of your C drive as time progresses in the instance, it ticks down at 0.05 gb per couple of minutes.

### Expected Behavior

The YouTube video player could have multiple playlists or multiple streams of the music mixes, making the dat files smaller but perhaps multiple instead, which allows the game to switch between more than one buffer as a file.

### Screenshots & Output Logs

![image](https://github.com/user-attachments/assets/1a27fa47-ea1c-492b-82f8-1f5792df2975)

### Are you in VR or Desktop mode?

Desktop

### Are you on Linux?

No

**Comment by owlboy** [comment]
I appreciate you taking the time to report this! This is something that should be reported to VRChat. I do not have control over the inner workings of video playback in Unity or VRChat. Nor do I use YouTube.

I'll also note that this is the first time in the 8 years that I have been running these events and that I have heard of this issue.

That 3.4GB file is the whole file. Nothing more should be downloaded. If more is being downloaded, then that is an internal bug that VRChat should address.

Based on your screenshot, it looks like only 3.4GB downloaded. It's 3.4GB up the tree. This leads me to believe that you and your friends are running with boot drives that have under 5GB of free space. This isn't great practice. But I'm not judging, I get how that happens. But it's not great practice. But like I said, things have worked exactly the same for many years, across countless events, and this is the first I have heard of this issue. This leads me to believe it is an issue with running your boot drive with out enough free space, or VRChat has a new bug that they need to investigate.

**Comment by mossarelli** [comment]
Yes, some people have 100 GB SSDs that still work, and with basic necessities installed, there is 7 GB free space, not even enough room for a 16 GB page file.

So when VRC and other apps fills up the disc space with temp cache files, it tends to get 0 GB really quick.

If it's not the implementation of the video player, but rather VRC's implementation of caching, then you can't fix it.

I'll just symlink the file onto the D drive and delete it later.

**Comment by sacred0** [comment]
Using the VRChat config file, you can customize the size and location of the VRChat cache. Unfortunately, I don't think the video player cache can be moved, but you can at least save some space by moving the VRChat cache. I recommend making the cache size fairly large, to avoid needing re-downloading things as often. https://docs.vrchat.com/docs/configuration-file
